glean-core/android/src/main/java/mozilla/telemetry/glean/private/PingType.kt

```diff
@@ -109,8 +112,10 @@ class PingType<ReasonCodesEnum> (
         }
         this.testCallback = null
 
-        val reasonString = reason?.let { this.reasonCodes[it.code()] }
-        this.innerPing.submit(reasonString)
```

Comment:
I wasn't actually sure whether to also dispatch the callback here?
I guess in tests it won't really matter much, because the time between calling submit and glean being initialized and thus running the glean-core submit should be small.

Reply:
RLB doesn't (because it doesn't have access to the glean_core dispatcher, and likely also for the reasons you've identified) so I'd keep it sync here for consistency.